### ISSUE TITLE
__str__ method in TabularCPD not working

### ISSUE THREAD

**Issue by yashu-seth** [pgmpy/pgmpy#484]
``` python
>>> from pgmpy.factors.CPD import TabularCPD 
>>> cpd = TabularCPD('grade', 3, [[0.1, 0.1],[0.1, 0.1],[0.8, 0.8]],evidence='evi1', evidence_card=2)
>>> print(cpd)
```

``` python
Traceback (most recent call last):
  File "<pyshell#19>", line 1, in <module>
    print(cpd)
  File "C:\Python34\lib\site-packages\pgmpy-0.1.0-py3.4.egg\pgmpy\factors\CPD.py", line 149, in __str__ row.extend(np.array(self.variables[row[0]])[indexes[:, i-1]].tolist())
TypeError: list indices must be integers, not str
```

**Comment by yashu-seth** [comment]
I need a little help with the tabulate function.
Please tell me if we can print header rows with unequal number of columns in a table or not?
It is not specifically mentioned in the the docs.

**Comment by ankurankan** [comment]
@yashu-seth I haven't worked with `tabulate`. Will check and let you know if I find anything.

**Comment by Erotemic** [comment]
I believe I fixed this issue in a recent patch.

**Comment by ankurankan** [comment]
Fixed in #524
